src/main/webapp/app/shared/user-settings/notification-settings/notification-settings.component.ts

```diff
@@ -42,10 +49,15 @@ export class NotificationSettingsComponent extends UserSettingsDirective impleme
         if (!foundSetting) {
             return;
         }
-        if (webApp) {
-            foundSetting!.webapp = !foundSetting!.webapp;
-        } else {
-            foundSetting!.email = !foundSetting!.email;
+        switch (communicationChannel) {
+            case NotificationSettingsCommunicationChannel.WEBAPP: {
+                foundSetting!.webapp = !foundSetting!.webapp;
```

Comment:
this is the toggling? might be worth a comment

Reply:
or for better readability:
````
let previousSetting = settingToUpdate!.webapp;
// toggle previous setting
settingToUpdate!.webapp = !previousSetting;

Reply:
I think the direct inverting is ok because this is quite common for booleans (at least from my experience). I added a comment